src/sentry/utils/http.py

```diff
@@ -218,5 +219,11 @@ def percent_encode(val: str) -> str:
     return quote(val).replace("%7E", "~").replace("/", "%2F")
 
 
-def is_using_customer_domain(request: HttpRequest) -> bool:
+class _HttpRequestWithSubdomain(HttpRequest):
+    """typing-only: to help with hinting for `.subdomain`"""
+
+    subdomain: str
+
+
+def is_using_customer_domain(request: HttpRequest) -> TypeGuard[_HttpRequestWithSubdomain]:
     return bool(hasattr(request, "subdomain") and request.subdomain)
```

Comment:
this allows safe access to `.subdomain` -- I'll probably go back and remove the mypy plugin for this and fix the accesses (basically require them to go through this helper function)